src/ffi/mod.rs

```diff
@@ -401,6 +415,15 @@ pub extern "C" fn telio_get_adapter_luid(dev: &telio) -> u64 {
     }
 }
 
+fn char_ptr_to_type<T: std::str::FromStr>(private_key: *const c_char) -> Result<T, telio_result> {
+    let cstr = unsafe { CStr::from_ptr(private_key) };
+    let private_key = cstr.to_str().map_err(|_| TELIO_RES_INVALID_STRING);
+    match private_key {
+        Ok(key) => key.parse().map_err(|_| TELIO_RES_INVALID_STRING),
```

Comment:
nit: Why not at least print the error that we ignore here (and 2 lines above)? This might make any debugging easier :)

Reply:
Doesn't adding `?` resolve that ? Since it is called in a ffi_try! macro which prints if the method returns an error.

Reply:
`ffi_try!` applies `?` to the result of `char_ptr_to_type`. As seen in this line the `parse()` error is ignored (bound to `_` and not used in any way) and always `TELIO_RES_INVALID_STRING` is used as a resulting error. This the `?` will print `TELIO_RES_INVALID_STRING`.